### ISSUE TITLE
Enhancement: OVN secondary network should have the possibility to disable mac spoof check

### ISSUE THREAD

**Issue by agonzalezrh** [ovn-kubernetes/ovn-kubernetes#3926]
Using `bridge` for secondary network allows to specify `macspoofchk` to enable or disable the mac spoofing check.

It would be useful for the OVN secondary network to have the same feature. I think a simply condition can be added here:

https://github.com/ovn-org/ovn-kubernetes/blob/c463ceae1c0ec5d260ed66583a5a0dd6948e7f7f/go-controller/pkg/ovn/base_network_controller_pods.go#L563

Use case: nested virtualization

**Comment by girishmg** [comment]
This should be easy to add. We have it downstream @cathy-zhou can you PTAL?

**Comment by agonzalezrh** [comment]
From my tests is needed as well to set "Addresses" to unknown.

**Comment by cathy-zhou** [comment]
the way we support it downstream is to add an annotation in the net-attach-def to indicate on this particular network, spoofcheck is disabled.

**Comment by maiqueb** [comment]
o/

could you elaborate on the design choice @cathy-zhou ? 

I.e. why have this a per-network attribute rather than per pod attachment ? 

Like ... assume everything connected to a network like this is "unsafe" ?

**Comment by agonzalezrh** [comment]
Do we have any new information about this?

**Comment by maiqueb** [comment]
I've been thinking again about this; it could make sense, and if the setting is done per network - and defaulted to having MAC spoofing - I am OK with it.

**Comment by agonzalezrh** [comment]
Thanks for the update, I will work on implement it and create a PR if you agree

**Comment by maiqueb** [comment]
> Thanks for the update, I will work on implement it and create a PR if you agree

Before you spend time on it, let's hear from another maintainer; @trozet / @jcaamano / @tssurya any thoughts about this feature ?

**Comment by trozet** [comment]
Yes, we are OK to support this. Seems like NAD is a natural place for the setting to live. Perhaps when we move NAD creation to a CRD it should be a field there.
